src/main/java/com/dev/movietheatre/dao/impl/CinemaHallDaoImpl.java

```diff
@@ -0,0 +1,48 @@
+package com.dev.movietheatre.dao.impl;
+
+import com.dev.movietheatre.dao.CinemaHallDao;
+import com.dev.movietheatre.exception.DataProcessingException;
+import com.dev.movietheatre.lib.Dao;
+import com.dev.movietheatre.model.CinemaHall;
+import com.dev.movietheatre.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class CinemaHallDaoImpl implements CinemaHallDao {
+    @Override
+    public CinemaHall add(CinemaHall cinemaHall) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(cinemaHall);
+            transaction.commit();
+            return cinemaHall;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert cinemaHall entity" + cinemaHall, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<CinemaHall> getAll() {
+        try {
+            Session session = HibernateUtil.getSessionFactory().openSession();
+            Query<CinemaHall> getAllCinemaHallsQuery = session.createQuery("from CinemaHall",
+                    CinemaHall.class);
+            return getAllCinemaHallsQuery.getResultList();
```

Comment:
You can return your list without local variable,but it`s up to you)

Reply:
thanks for advice